l10n_ro_siruta/models/res_partner.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# ©  2015 Forest and Biomass Services Romania
+# See README.rst file on addons root folder for license details
+
+from openerp import models, fields, api
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.onchange('city_id')
+    def _onchange_city_id(self):
+        if self.city_id:
+            self.commune_id = self.city_id.commune_id.id
+            self.state_id = self.city_id.state_id.id
+            self.zone_id = self.city_id.zone_id.id
+            self.country_id = self.city_id.country_id.id
+
+    city_id = fields.Many2one('res.country.city',
```

Comment:
You should probably add this field in an override of res.partner::_address_fields(), which facilitates the address sync between partners and contacts that have 'use company address' set. Same for commune, probably.